src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocations.Fixer.cs

```diff
@@ -128,30 +127,8 @@ Task<Document> CreateChangedDocumentAsync(CancellationToken _)
 
             newArguments = newArguments.Add(cancellationTokenArgument);
 
-            SyntaxNode newInstance;
-            // The instance is null when calling a static method from another type
-            switch (invocation.Instance)
-            {
-                case null:
-                    newInstance = expression;
-                    break;
-                case IConditionalAccessInstanceOperation:
-                    newInstance = GetConditionalOperationInvocationExpression(invocation.Syntax);
-                    break;
-                default:
-                    if (invocation.Instance.IsImplicit)
-                    {
-                        newInstance = invocation.GetInstanceSyntax()!;
-                    }
```

Comment:
> _Adds a test case for a static generic method, which was green before applying the fix, to guard against regressions in that behavior_

The CI is still passing because apparently I forgot to add unit tests for all three cases you removed.

- It seems we don't have a conditional operation case in the unit tests. Can you please add one?
- I'm trying to remember what `IsImplicit` is referring to. It's either:
    - An invocation of a method from an explicit interface implementation, or
    - An invocation of a method passed as a callback to another method (a `Func` or an `Action`).
  Independently if it's one or the other, we don't have unit tests for any of them. Would you mind adding one to make sure that functionality is not regressed?

Thanks for adding a method for the static case.

Reply:
From the [IConditionalAccessInstanceOperation Interface docs](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.operations.iconditionalaccessinstanceoperation?view=roslyn-dotnet) it sounds like you're referring to a call like `someExpression?.Method()`. Isn't that covered by [CS_Diagnostic_DereferencePossibleNullReference](https://github.com/dotnet/roslyn-analyzers/blob/main/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocationsTests.cs#L1974-L2018)?

`IsImplicit` seems to refer to the "this" object passed to the method, either the instance or null if it's a static method. In any case, when I tried adding this block to the fixer:

```csharp
if (invocation.Instance != null && invocation.Instance.IsImplicit)
{
    throw new Exception("Implicit");
}
```

…the result was that 73 tests failed. First one was [CS_Diagnostic_ActionDelegateAwait](https://github.com/dotnet/roslyn-analyzers/blob/main/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocationsTests.cs#L1146-L1180) where `MethodAsync()` is being called without `this.`, which seems to track. So I think it's safe to say that test case is covered.

While the removal of that code looks scary, I think it makes sense philisophically. The fixer is only trying to adjust what's going on in the ArgumentList. So trying to rebuild the part of the invocation "outside" the parens from scratch based on a number of conditions doesn't really make sense when `expression` is already going to contain whatever that outside part was originally highlighted by the red squiggly.

To illustrate that point further, I added 2 additional tests `CS_Diagnostic_NullCoalescedDelegates` and `CS_Diagnostic_NullCoalescedDelegatesWithInvoke`, both of which have 2 delegates and call either `(f1 ?? f2)()` or `(f1 ?? f2).Invoke()`. The current version on `main` will rewrite `(f1 ?? f2)()` to `(f1 ?? f2).Invoke(c)`, adding the explicit `Invoke` where none was there before. I assume it's preferable for a fixer to do the bare minimum to implement the fix without changing any other code, and this PR respects whether or not the explicit `Invoke` is used in the expression.